source/val/construct.cpp

```diff
@@ -64,4 +66,53 @@ const BasicBlock* Construct::exit_block() const { return exit_block_; }
 BasicBlock* Construct::exit_block() { return exit_block_; }
 
 void Construct::set_exit(BasicBlock* block) { exit_block_ = block; }
+
+Construct::ConstructBlockSet Construct::blocks(Function* function) const {
+  auto header = entry_block();
+  auto merge = exit_block();
+  assert(header && merge);
```

Comment:
It's a little thing, but I would have done two asserts, so it's easier to see what the failure was: missing header? or missing merge.

Not a request to change this.